multicluster/controllers/multicluster/node_controller.go

```diff
@@ -73,65 +78,215 @@ func NewNodeReconciler(
 
 func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	klog.V(2).InfoS("Reconciling Node", "node", req.Name)
+	if !r.initialized {
+		if err := r.initialize(); err != nil {
+			return ctrl.Result{}, err
+		}
+		r.initialized = true
+	}
 	gw := &mcsv1alpha1.Gateway{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      req.Name,
 			Namespace: r.namespace,
 		},
 	}
-	// When the Node is annotated with 'multicluster.antrea.io/gateway=true' as a Gateway:
-	//   - Delete the Gateway if the Node is deleted
-	//   - Update the Gateway if Node's InternalIP or GatewayIP is updated
-	//   - Create a new Gateway if there is no existing Gateway
+
 	node := &corev1.Node{}
 	if err := r.Client.Get(ctx, req.NamespacedName, node); err != nil {
 		if apierrors.IsNotFound(err) {
-			err := r.Client.Delete(ctx, gw, &client.DeleteOptions{})
-			return ctrl.Result{}, client.IgnoreNotFound(err)
+			if r.activeGateway == req.Name {
+				if err := r.recreateActiveGateway(ctx, gw); err != nil {
+					return ctrl.Result{}, err
+				}
+				return ctrl.Result{}, nil
+			}
 		}
 		klog.ErrorS(err, "Failed to get Node", "node", req.Name)
-		return ctrl.Result{}, err
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	_, isGW := node.Annotations[common.GatewayAnnotation]
-	var err error
-	gwNamespacedName := types.NamespacedName{
-		Name:      node.Name,
-		Namespace: r.namespace,
+	_, hasGWAnnotation := node.Annotations[common.GatewayAnnotation]
```

Comment:
I feel strange to operate against a possibly empty Node struct.

Reply:
Refined.